clouddriver-ecs/src/test/groovy/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperationSpec.groovy

```diff
@@ -1853,6 +1853,39 @@ class CreateServerGroupAtomicOperationSpec extends CommonAtomicOperation {
     result.getExecutionRoleArn() == "arn:aws-us-gov:iam:123123123123:role/test-role"
   }
 
+  def 'should return valid task def from spelProcessedArtifact'() {
+    given:
+    def credentials = Mock(NetflixAssumeRoleAmazonCredentials) {
+      getName() >> { "test" }
+      getRegions() >> { [new AmazonCredentials.AWSRegion('us-west-1', ['us-west-1a', 'us-west-1b'])] }
+      getAssumeRole() >> { 'arn:aws-us-gov:iam:123123123123:role/test-role' }
+      getAccountId() >> { 'test' }
+    }
+    def resolvedArtifact = createResolvedArtifact()
```

Comment:
what happens if we don't mock this? the resolved artifact is only used when we need to download the file, so running this test w/o would better ensure we're actually using the spelProcessedTaskDefinitionArtifact